### DISCUSSION TITLE
I'm trying to use edge attributes in a GATConv network. I'm facing issue when adding self_loop.

### DISCUSSION THREAD

**Discussion by arake001** [pyg-team/pytorch_geometric#6821] [Q&A]
I am facing issues with the GATConv when using edge_attributes. 

I tried using single-dimension edge_attribute and multiple-dimension edge attributes.

Example graph structure

Data(edge_attr=[13848], edge_index=[2, 13848], x=[2295, 100], y=[1])

Data(edge_attr=[13848,1], edge_index=[2, 13848], x=[2295, 100], y=[1]) 

Data(edge_attr=[13848, 2], edge_index=[2, 13848], x=[2295, 100], y=[1])

The following is the error.

 File "/home/ankith/graph_files/cvpr/casme2/pytorch_geometric/torch_geometric/nn/conv/edge_gat.py", line 289, in forward
    edge_index, edge_attr = add_self_loops(
  File "/home/ankith/anaconda3/envs/gconv-cuda/lib/python3.8/site-packages/torch_geometric/utils/loop.py", line 89, in add_self_loops
    loop_weight = edge_weight.new_full((N, ), fill_value)
TypeError: new_full(): argument 'fill_value' (position 2) must be Number, not str

Any clue how to solve this issue?

 Thank you in advance!

**Comment by rusty1s** [comment]
Can you show me how you call `add_self_loops` in `edge_gat.py`?

**Reply by arake001** [reply]
I haven't done any changes to the add_self_loop section of the GATConv layer.

The other thing I noticed is that the code works on the CPU version of PyTorch 1.10 and when I try the same code on the GPU version of PyTorch 1.6, it fails. 

Is the issue with the version of the Pytorch or when using GPU trigger an error?

**Reply by rusty1s** [reply]
Mh, PyTorch 1.6 is now over two years old, so we don't give any guarantees that latest PyG works with it. I would recommend updating your PyTorch version.
